lib/src/builder/codegen/accessors_generator.dart

```diff
@@ -352,6 +364,15 @@ abstract class TypedMapAccessorsGenerator extends BoilerplateDeclarationGenerato
 
     output.write(staticVariablesImpl);
 
+    if (requiredPropChecks.isNotEmpty) {
+      final validateRequiredPropsMethod = '\n  @override\n'
+          '  @mustCallSuper\n'
+          '  void validateRequiredProps() {\n'
```

Comment:
I'm not sure if this was supposed to call `super.validateRequiredProps()`, but I get this error in generated files when I add that `error: The method 'validateRequiredProps' is always abstract in the supertype.`

Reply:
I tried adding this in locally to look more closely at the issues, and it looks like this only affects legacy Dart-2-only boilerplate...

I tried seeing if we could add a super-constraint to those mixins:
```diff
diff --git a/lib/src/builder/codegen/accessors_generator.dart b/lib/src/builder/codegen/accessors_generator.dart
index 1003b2ed..f19daee3 100644
--- a/lib/src/builder/codegen/accessors_generator.dart
+++ b/lib/src/builder/codegen/accessors_generator.dart
@@ -82,9 +82,7 @@ abstract class TypedMapAccessorsGenerator extends BoilerplateDeclarationGenerato
       final implementsClause = 'implements ${names.consumerName}$typeParamsOnSuper';
       generatedClass
         ..write(generatedCodeUseOnlyDeprecation)
-        ..writeln('abstract class $accessorsMixinName$typeParamsOnClass $implementsClause {\n'
-            '  @override'
-            '  Map get ${type.isProps ? 'props' : 'state'};\n');
+        ..writeln('mixin $accessorsMixinName$typeParamsOnClass on ${type.isProps ? 'UiProps' : 'UiState'} $implementsClause {\n');
       if (type.isMixin) {
         generatedClass.writeln(_generateStaticMetaDecl(names.publicName, type.isProps));
         generatedClass.write(_copyStaticMembers(node));
@@ -367,6 +365,7 @@ abstract class TypedMapAccessorsGenerator extends BoilerplateDeclarationGenerato
       final validateRequiredPropsMethod = '\n  @override\n'
           '  @mustCallSuper\n'
           '  void validateRequiredProps() {\n'
+          '    super.validateRequiredProps();\n'
           '    ${requiredPropChecks.join('\n')}\n'
           '  }\n';
       output.write(validateRequiredPropsMethod);
```
...but that caused a whole bunch of other typing issues that don't seem feasible to fix without a lot of effort and potential for regressions.

So I'm thinking we just don't support required prop validation for that boilerplate version; that boilerplate is really only used within over_react, so I don't think that should be a huge issue.

With these changes, it seems to work!
```diff
diff --git a/lib/src/builder/codegen/accessors_generator.dart b/lib/src/builder/codegen/accessors_generator.dart
index 1003b2ed..c388384a 100644
--- a/lib/src/builder/codegen/accessors_generator.dart
+++ b/lib/src/builder/codegen/accessors_generator.dart
@@ -363,10 +363,11 @@ abstract class TypedMapAccessorsGenerator extends BoilerplateDeclarationGenerato
 
     output.write(staticVariablesImpl);
 
-    if (requiredPropChecks.isNotEmpty) {
+    if (type.isProps && version != Version.v3_legacyDart2Only) {
       final validateRequiredPropsMethod = '\n  @override\n'
           '  @mustCallSuper\n'
           '  void validateRequiredProps() {\n'
+          '    super.validateRequiredProps();\n'
           '    ${requiredPropChecks.join('\n')}\n'
           '  }\n';
       output.write(validateRequiredPropsMethod);
```

Also, I'm thinking it'd probably be safer to not conditionally output the override based on `requiredPropChecks.isNotEmpty`. If we always output it, then it'll help our tests catch any cases where the generated code is valid when there are no required props, but might become invalid if required props are ever added. And I'd expect those empty overrides, where it just calls super, to get compiled out in dart2js.

Reply:
Oh nice! I didn't realize it was just for some boilerplates! I added that change in [8890aad](https://github.com/Workiva/over_react/pull/862/commits/8890aad7c2104f68200e310a12ac52be49c07501), but I'm still getting a few `error: The method 'validateRequiredProps' is always abstract in the supertype.` Like [this one](https://github.com/Workiva/over_react/blob/c346c0c371957eceb59fefafb0750c9017c1cb3d/test/over_react/component/fixtures/dummy_component.over_react.g.dart#L51) - what boilerplate version is that?

Reply:
Oh that's the v2 backwards compat boilerplate. Hmm it looks like that version has the same issues... 

Dang, sorry about that, I could have sworn it all looked good locally when I was testing it out; maybe my analyzer wasn't catching up? Or maybe I had other local generation changes I forgot to comment out?

Hmm tweaking the generation of that specific case in legacy boilerplate, so that the accessors mixin is a mixin on the base class instead of implementing it, seems to fix the issue:
```diff
diff --git a/lib/src/builder/codegen/accessors_generator.dart b/lib/src/builder/codegen/accessors_generator.dart
index 3d72fb35..85be0eae 100644
--- a/lib/src/builder/codegen/accessors_generator.dart
+++ b/lib/src/builder/codegen/accessors_generator.dart
@@ -79,16 +79,20 @@ abstract class TypedMapAccessorsGenerator extends BoilerplateDeclarationGenerato
       }
     }
     if (version.isLegacy) {
+      if (type.isMixin) {
         final implementsClause = 'implements ${names.consumerName}$typeParamsOnSuper';
         generatedClass
           ..write(generatedCodeUseOnlyDeprecation)
           ..writeln('abstract class $accessorsMixinName$typeParamsOnClass $implementsClause {\n'
               '  @override'
               '  Map get ${type.isProps ? 'props' : 'state'};\n');
-      if (type.isMixin) {
         generatedClass.writeln(_generateStaticMetaDecl(names.publicName, type.isProps));
         generatedClass.write(_copyStaticMembers(node));
         generatedClass.write(_copyClassMembers(node));
+      } else {
+        generatedClass
+          ..write(generatedCodeUseOnlyDeprecation)
+          ..writeln('mixin $accessorsMixinName$typeParamsOnClass on ${names.consumerName}$typeParamsOnSuper {\n');
       }
     } else {
       generatedClass
```
Though, I think that would only fix it for props declared in props classes, and not mixins... (that `if (type.isMixin)` case was the one I had issues with [here](https://github.com/Workiva/over_react/pull/862/commits/c346c0c371957eceb59fefafb0750c9017c1cb3d#r1420812787)).

I'm starting to think we should just not support this validation for either of the legacy boilerplates :/. How does that sound to you?

Reply:
Not supporting validation for legacy boilerplates works for me! I think some of them already throw errors for required props anyway